### ISSUE TITLE
Indent is disconnecting events (ms-access)

### ISSUE THREAD

**Issue by Doev** [rubberduck-vba/Rubberduck#4421]
Hi,

I recognize this behaviour some times now, but finally I see the connection to the rubberduck addon. I can't reproduce the bug but this is what happend yesterday:

1) I pressed the "pending" button to analyze the project
2) I do a "indent" on the whole project

After that all event methods of all controls in all forms where disconnected. That means, that the vba-code was still in the modules, but in the visual designer, the event property was unset.

I remember that similar happend in the past, but I never saw rubberduck as the problem.

**Comment by retailcoder** [comment]
I presume Access forms are using member attributes to wire up with macros; if that's the case then rewriting the entire module would indeed lose the member attributes - this is a known issue and a rather hard problem to graciously handle, since the member attributes aren't visible in the VBE code panes: restoring them would require re-importing the modified modules.

Avoid indenting (or otherwise rewriting - this includes refactorings and inspection quickfixes) modules that have member attributes.

Note: ActiveX controls use a different interface-based wiring mechanism that isn't affected by this.

**Comment by bclothier** [comment]
No it doesn't use member attributes. Whether an event will fire is governed by another property of the form. In this example, I set the `OnCurrent` (displayed as `On Current`) property to string `[Event Procedure]`:
![image](https://user-images.githubusercontent.com/2367644/46668180-587bb980-cb91-11e8-994c-8a3d873dcdfa.png)

This wires up the event accordingly (and I added `Debug.Print "hi"`):
![image](https://user-images.githubusercontent.com/2367644/46668271-88c35800-cb91-11e8-8b5c-e522efa38f17.png)

Now, if I were to delete the entire procedure, it automatically clears the `OnCurrent` property:
![image](https://user-images.githubusercontent.com/2367644/46668332-adb7cb00-cb91-11e8-987e-4cb80c1958dd.png)

Likewise, when I use VBE's dropdowns to create the event handler... As soon as I select `Form` from the left-hand dropdown, `Load` event is selected by default, a empty stub created and the `OnLoad` property is then updated:
![image](https://user-images.githubusercontent.com/2367644/46668390-e5267780-cb91-11e8-95e5-0c79baea3890.png)

So, given this unindented code:
![image](https://user-images.githubusercontent.com/2367644/46668462-1bfc8d80-cb92-11e8-8aa3-9bf86bacd30b.png)

Indenting the module works and the property is still preserved:
![image](https://user-images.githubusercontent.com/2367644/46668792-266b5700-cb93-11e8-9401-0c7428a78400.png)

Note, though, that pressing ctrl+z to undo will lose the properties:
![image](https://user-images.githubusercontent.com/2367644/46668844-4569e900-cb93-11e8-96fa-84ff97af6cb0.png)

But redoing it restores it as expected. 

Do note that there has been known cases of the event properties becoming disconnected from VBA codebase in versions prior to Access 2010. Since Access 2010, the synchronization has been pretty robust. 

I tried few weird indenting or different indenting commands but in no case could I reproduce the issue easily.

**Comment by retailcoder** [comment]
@bclothier so this is a no-repro for now?

**Comment by Doev** [comment]
I can reproduce the problem. If you start indent while the app is in debug mode, the events are disconnected.

1) create new Access Db
2) create a Form
3) create a Button (cancel the wizard)
4) create a click event on the button with one line of code: "stop"
5) go to Formview and press the button // VBE opens -> Debug mode
6) press pending
7) indent the project (maybe just the procedure works too)
8) the event is no more connected

**Comment by bclothier** [comment]
Hmm. I followed the same steps and I can't reproduce this. You didn't say which version of Access. Is it 2010 or newer?

**Comment by Doev** [comment]
2016

**Comment by bclothier** [comment]
I just tested on O365's 2016. Still can't reproduce it using same steps above... 😕  

Can you give us the Rubberduck version? Maybe logs, too?

**Comment by Doev** [comment]
Windows 7 Microsoft Office Professional Pro 2016
I have done some tests. Only if the App is stopped in Debug mode, the events are unwired.

Version 2.2.6728.31668
OS: Microsoft Windows NT 6.1.7601 Service Pack 1, x64
Host Product: Microsoft Office 2016 x86
Host Version: 16.0.4711.1000
Host Executable: MSACCESS.EXE

**Comment by bclothier** [comment]
Bleh, I'm retarded. I did *not* follow steps. I missed the part about changing to normal view and actually clicking button to enter the break mode. 

So yes, I can reproduce. However, I also get the dialog asking me if I'd want to reset the project, which is expected given the command. 

With that clarified, I'm inclined to say that this is going to be by design because we really should not be running a majority of Rubberduck commands while in break mode since it's most likely to reset the project's state. And resetting the project's state is something of a no-no in my book, especially when you have open forms since those get their states reset and who know what else wil happen. 

Need moar caffine....

**Comment by Doev** [comment]
Same on this PC

Version 2.2.0.3918
OS: Microsoft Windows NT 6.1.7601 Service Pack 1, x64
Host Product: Microsoft Office 2016 x64
Host Version: 16.0.4711.1000
Host Executable: MSACCESS.EXE

**Comment by Doev** [comment]
yes, after knowing the problem, it is not a big problem anymore. Maybe a test for "beeing in debug mode" when the user press indent, would be fine.
